### PR TITLE
Refactor project structure and module organization

### PR DESCRIPTION
**Pull request by bniladridas**
Restructures project modules and workflows for improved clarity and consistency. Consolidates related functionality and standardizes naming conventions.

## Changes Made
- Renamed modules for better semantic clarity
- Reorganized workflow files with shorter names
- Added comprehensive test suite for Python analysis
- Updated all module imports and references
- Enhanced README documentation

## Module Restructuring
**Renamed:**
- `src/constraint_solver/` → `src/solver/`
- `src/runtime_tracer/` → `src/tracer/`

**Workflow Files:**
- `.github/release-drafter.yml` → `.github/drafter.yml`
- `.github/workflows/release_notes.yml` → `.github/workflows/notes.yml`

## Test Suite Addition
**New test files:**
- `tests/sample.py` - Basic function examples for analysis
- `tests/classes.py` - Class and method examples
- `tests/complex.py` - Complex type scenarios and edge cases

## Impact
- Improves code organization and maintainability
- Provides comprehensive test coverage for Python analysis
- Standardizes module naming conventions
- Reduces workflow file name verbosity
- Maintains backward compatibility in functionality

## Testing
- All module imports updated and verified
- Workflow files tested for correct functionality
- Test files provide comprehensive coverage scenarios

## Type of Change
- Code refactoring and reorganization
- Test suite enhancement
- No breaking changes to public API

### CODE REVIEW

File: src/tracer/mod.rs [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+
+    #[test]
+    fn test_tracer_initialization() {
+        let tracer = RuntimeTracer::new(false);
+        assert!(!tracer.verbose);
+    }
```
[the remark this answers is not in the document]
Comment (bniladridas) [reply src/tracer/mod.rs]:
```
#[cfg(test)]
mod tests {
    use super::*;
    use std::fs;
    use tempfile::NamedTempFile;  // This line is present
    
    // ... rest of the test code ...
}
```
